File: LeePoongHo/TargetNumber.java
class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        int index = 0;
        int number = 0;
        
        return tcount(index, numbers, number, target); 
    }
    
    public int tcount(int index, int[] numbers, int number, int target) {
        int length = numbers.length-1;
        
        if(index == length) {
            if(number+numbers[index] == target) {
                return 1;
            } else if(number-numbers[index] == target) {
                return 1;
            } else {
                return 0;
            }
        } else {
            return tcount(index+1, numbers, number+numbers[index], target)+tcount(index+1, numbers, number-numbers[index], target);
        }
    }
}

File: sjjeong/TargetNumber.kt
class Solution {
    fun solution(numbers: IntArray, target: Int): Int {
        val diff = (numbers.sum() - target) / 2
        return find(numbers, diff, 0, numbers.size)
    }

    fun find(numbers: IntArray, diff: Int, index: Int, size: Int): Int {
        var answer = 0
        for (i in index until size) {
            val number = numbers[i]
            if (number == diff) {
                answer++
            } else if (number < diff) {
                answer += find(numbers, diff - number, i + 1, size)
            }
        }
        return answer
    }
}